apps/meteor/app/file-upload/server/methods/sendFileMessage.ts

```diff
@@ -22,13 +24,48 @@ import { FileUpload } from '../lib/FileUpload';
 
 function validateFileRequiredFields(file: Partial<IUpload>): asserts file is AtLeast<IUpload, '_id' | 'name' | 'type' | 'size'> {
 	const requiredFields = ['_id', 'name', 'type', 'size'];
-	requiredFields.forEach((field) => {
+	for (const field of requiredFields) {
 		if (!Object.keys(file).includes(field)) {
 			throw new Meteor.Error('error-invalid-file', 'Invalid file');
 		}
-	});
+	}
 }
 
+const logger = new Logger('sendFileMessage');
+
+type parseMultipleFilesIntoMessageAttachmentsResult = { files: FileProp[]; attachments: MessageAttachment[] };
+export const parseMultipleFilesIntoMessageAttachments = async (
+	filesToConfirm: Partial<IUpload>[],
+	roomId: string,
+	user: IUser,
+): Promise<parseMultipleFilesIntoMessageAttachmentsResult> => {
+	const result: parseMultipleFilesIntoMessageAttachmentsResult = { files: [], attachments: [] };
+
+	await Promise.all(
+		filesToConfirm.reduce<Array<Promise<void>>>((acc, file) => {
+			if (!file) return acc;
+
+			acc.push(
+				(async () => {
+					try {
+						const { files, attachments } = await parseFileIntoMessageAttachments(file, roomId, user);
+						result.files.push(...files);
+						result.attachments.push(...attachments);
+					} catch (error) {
+						// Not an important error, it should not happen and if it happens wil affect the attachment
+						// preview in the message object only
+						logger.warn('Error processing file:', file, error);
+					}
+				})(),
+			);
+
+			return acc;
+		}, []),
+	);
+
+	return result;
```

Comment:
also having a const that is going to be mutated, a promise awaited that will cause mutations inside, auto invoked functions all this together, makes me think into a different solution, something like 
```suggestion
	const parsed = await Promise.all(
		filesToConfirm.filter(Boolean).map(file) => {
			try {
				return await parseFileIntoMessageAttachments(file, roomId, user);
			} catch (error) {
				// Not an important error, it should not happen and if it happens wil affect the attachment
				// preview in the message object only
				logger.warn('Error processing file:', file, error);
			}
			return { files: [], attachments: [] };
		})
	);
	return { files: parsed.flatMap(({ file }) => file), attachments: parsed.flatMap(({ attachments }) => attachments) }
	};
```